backend/executor.go

```diff
@@ -121,12 +124,15 @@ func (executor *grpcExecutor) ExecuteOrchestrator(ctx context.Context, iid api.I
 	}
 
 	// Wait for the connected worker to signal that it's done executing the work-item
-	// TODO: Timeout logic - i.e. handle the case where we never hear back from the remote worker (due to a hang, etc.).
 	select {
 	case <-ctx.Done():
 		executor.logger.Warnf("%s: context canceled before receiving orchestrator result", iid)
 		return nil, ctx.Err()
 	case <-result.complete:
+		executor.logger.Debug("orchestrator", iid, "got result")
+		if result.Response == nil {
+			return nil, errors.New("operation aborted")
```

Comment:
Is this new return statement the main part of the fix?

Reply:
Yes, now if the operation is aborted, we get a signal in `result.complete`, but the `Response` property will be nil. This is important because we need to unblock this `select` block